halyard-config/src/main/java/com/netflix/spinnaker/halyard/config/services/v1/ProviderService.java

```diff
@@ -42,21 +48,32 @@ public Provider getProvider(String deploymentName, String providerName) {
       provider = providers.getDockerRegistry();
     } else if (providerName.toLowerCase().equals("google")) {
       provider = providers.getGoogle();
+    } else {
```

Comment:
This is leaving once the `NodeFilter` is used to do the lookup here - I agree it's messy especially when I have the more powerful filters to rely on.